### ISSUE TITLE
PgWire: compatibility with python's asyncpg Postgres library

### ISSUE THREAD

**Issue by RiccardoGiro** [questdb/questdb#1100]
Does QuestDB support [asyncpg](https://github.com/MagicStack/asyncpg) library?

**Comment by snenkov** [comment]
Hey @RiccardoGiro are you looking at this library for reading from QDB or for writing to QDB (or both)?
For writing the ILP is quite fast already I believe - hence my question.

**Comment by RiccardoGiro** [comment]
Hello @snenkov, I am still trying to figure this out. Here is the full framework: I am developing a Python app to repurpose legacy datasets (tens of TB stored in a proprietary and messy binary format). My software currently reads chunks of data and organizes them into a Pandas dataframe. I now need to find a way to:

1. Efficiently bulk insert each dataframe. Saving a dataframe as .csv and then importing it to QuestDB is extremely inefficient (the Pandas -> .csv part takes an incredible amount of time);
2. Reading data from QuestDB and storing the output of the query in a dataframe.

I was thinking to use asyncpg (at least when reading from the database) because it employs a binary input/output protocol, therefore there is no text encoding->decoding when data are retrieved from QuestDB. This is a typical bottleneck of other PostgreSQL-based clients (e.g., psycopg) which read from the database and transfer every data type as text. 

Thanks for helping!

**Comment by ideoma** [comment]
QuestDB supports most of Postgres Wire protocol not only psycorpg. Many other programming languages and their Postgres drivers were tested and work successfully (Rust, Java, C#, NodeJS etc). It's hard to tell without trying if asyncpg will work but it is very likely that it will.

**Comment by ideoma** [comment]
I gave it a spin and it didn't go well

I tried

```
import asyncio
import asyncpg

async def run():
    conn = await asyncpg.connect(user='admin', password='quest',
                                 database='qdb', host='127.0.0.1', port=8812, command_timeout=2,ssl=False)
    query = await conn.prepare('tables')
    values = await query.fetch()
    print(values)

loop = asyncio.get_event_loop()
loop.run_until_complete(run())

```

Connection goes ok but `prepare()` (or `fetch()` without `prepare()`) hangs so there is protocol feature QuestDB does not support yet

**Comment by RiccardoGiro** [comment]
Thank you @ideoma for the quick test. I will therefore try to follow the kind suggestion of @snenkov and use ILP for writing + Rest API for reading. I will let you know how it goes!

**Comment by ideoma** [comment]
Fixed in #1751
